docs(readme): Rewrite to prepare for release (#296)
eitsupi/neo-r-polars: README.Rmd

)
```

# polars _(R Polars)_

<!-- TODO: add link to discord -->

<!-- badges: start -->
[![R-multiverse status](https://img.shields.io/badge/dynamic/json?url=https%3A%2F%2Fcommunity.r-multiverse.org%2Fapi%2Fpackages%2Fpolars&query=%24.Version&label=r-multiverse)](https://community.r-multiverse.org/polars)
[![R-universe status badge](https://rpolars.r-universe.dev/badges/polars)](https://rpolars.r-universe.dev)
[![CRAN status](https://www.r-pkg.org/badges/version/polars)](https://CRAN.R-project.org/package=polars)
[![Docs dev version](https://img.shields.io/badge/docs-dev-blue.svg)](https://pola-rs.github.io/r-polars)
<!-- badges: end -->

Polars is a blazingly fast DataFrame library for manipulating structured data.
The core is written in Rust.

> The goal of Polars is to provide a lightning fast DataFrame library that:

Author: The "ligthning fast" and "blazingly fast" are duplicated here. How about:

```
*badges*

Polars is a blazingly fast DataFrame library, written in Rust. Its goal
is to deliver fast, efficient data processing that:
```

- Utilizes all available cores on your machine.
- Optimizes queries to reduce unneeded work/memory allocations.
- Handles datasets much larger than your available RAM.

> - A consistent and predictable API.

Author: ```suggestion
- Follows a consistent and predictable API.
```

- Adheres to a strict schema (data-types should be known before running the query).

This `polars` R package provides the R bindings for Polars.

> It can be used to convert R DataFrames to Polars DataFrames and vice versa,
> as well as to integrate with other common R packages.

Author: I feel it's a bit reductive to mention only conversion between R and Polars. How about this?

```suggestion
It can be used to process Polars DataFrames and LazyFrames, convert objects 
between Polars and R, and can be integrated with other common R packages.
```

Reviewer: Again, I'm not sure if the term LazyFrame is appropriate here; I believe LazyFrame is commonly referred to as a Query Plan or something similar.

Reviewer: How about 54ad963d1502264065b3d13a8cabddf5eddd2623?

Author: It appears many times in the user guide, even when introducing the lazy api: https://docs.pola.rs/user-guide/lazy/

There's even a function called LazyFrame so I don't see a problem with that.

Author: > How about [54ad963](https://github.com/eitsupi/neo-r-polars/commit/54ad963d1502264065b3d13a8cabddf5eddd2623)?

Fine for  me, I just don't want to replace all "LazyFrame" in docs by query plans.

)
```

# polars _(R Polars)_

<!-- TODO: add link to discord -->

<!-- badges: start -->
[![R-multiverse status](https://img.shields.io/badge/dynamic/json?url=https%3A%2F%2Fcommunity.r-multiverse.org%2Fapi%2Fpackages%2Fpolars&query=%24.Version&label=r-multiverse)](https://community.r-multiverse.org/polars)
[![R-universe status badge](https://rpolars.r-universe.dev/badges/polars)](https://rpolars.r-universe.dev)
[![CRAN status](https://www.r-pkg.org/badges/version/polars)](https://CRAN.R-project.org/package=polars)
[![Docs dev version](https://img.shields.io/badge/docs-dev-blue.svg)](https://pola-rs.github.io/r-polars)
<!-- badges: end -->

Polars R bindings.

Polars is a blazingly fast DataFrame library for manipulating structured data. The core is written
in Rust, and available for Python, R and NodeJS.

- **Fast**: Written from scratch in Rust, designed close to the machine and without external
  dependencies.

> - **I/O**: First class support for all common data storage layers: local & cloud storage.

Reviewer: I think this works after #354, but Polars cannot read from public blob storage without credentials, so without an account I cannot test.
(Checking the upstream object-store, it looks like there is functionality for anonymous access, if the upstream Polars is modified, anonymous access should be allowed, but I didn't have time to create a bug report.)

)
```

# polars

<!-- TODO: add link to discord -->

<!-- badges: start -->
[![R-multiverse status](https://img.shields.io/badge/dynamic/json?url=https%3A%2F%2Fcommunity.r-multiverse.org%2Fapi%2Fpackages%2Fpolars&query=%24.Version&label=r-multiverse)](https://community.r-multiverse.org/polars)
[![R-universe status badge](https://rpolars.r-universe.dev/badges/polars)](https://rpolars.r-universe.dev)
[![CRAN status](https://www.r-pkg.org/badges/version/polars)](https://CRAN.R-project.org/package=polars)
[![Docs dev version](https://img.shields.io/badge/docs-dev-blue.svg)](https://pola-rs.github.io/r-polars)
<!-- badges: end -->

## Overview

Polars is a DataFrame interface on top of an OLAP Query Engine implemented in Rust using
[Apache Arrow Columnar Format](https://arrow.apache.org/docs/format/Columnar.html) as the memory

> model.

Author: I know this is from the README of Python Polars but to me it's just confusing.

I think it's very likely that an average R doesn't know what an OLAP query engine or the Arrow columnar format is (I don't even know what OLAP means). 

I don't think we should have this to start the README. The current r-polars README is more user-friendly IMO (although we don't have to list all kinds of operations Polars can perform).

Reviewer: I understand what you are saying, but I don't know if the terminology difficulty justifies leaving this place, given that PyPI[^pypi], crates.io[^cratesio], or npm[^npm] all show something like this in their READMEs.

[^pypi]: <https://pypi.org/project/polars/>
[^cratesio]: <https://crates.io/crates/polars>
[^npm]: <https://www.npmjs.com/package/nodejs-polars>

Author: I don't think an average user goes to either of those sites. They go straight to the official documentation website which is much more user friendly (https://docs.pola.rs/).

I really think we should drop this paragraph and keep the one on current r-polars: it has no value added for the average user, and the experienced ones can easily find this info elsewhere.

Reviewer: I see. How about I copied it from the index page of the user guide?

- Lazy | eager execution
- Multi-threaded
- SIMD
- Query optimization
- Powerful expression API
- Hybrid Streaming (larger-than-RAM datasets)

> - Rust | Python | NodeJS | R | ...

Author: Similarly, I expect most average R users to not be knowledgeable about multithreading, SIMD, query optimization, etc.

This `{polars}` R package provides the R bindings for Polars.
It can be used to convert R DataFrames to Polars DataFrames and vice versa,
as well as to integrate with other common R packages.

To learn more, read the [online documentation](https://pola-rs.github.io/r-polars/) for this R package,
and the [user guide](https://docs.pola.rs/) for Python / Rust Polars.

## Installation

The recommended way to install this package is from the R-multiverse community repository:

```r
Sys.setenv(NOT_CRAN = "true")
install.packages("polars", repos = "https://community.r-multiverse.org")
```

## Usage

To avoid conflicts with other packages and base R function names, `{polars}`'s
top level functions are hosted in the `pl` environment, and accessible via the
`pl$` prefix.

> This means that `polars` queries written in Python and in R are very similar.

Author: Nitpick but we should decide if we use `polars` or `{polars}` in the README (and in the docs in general).

(I personally prefer `polars` just to avoid extra typing)

Reviewer: Only use `` `polars` `` now.

> For example, rewriting the Python example from <https://github.com/pola-rs/polars> in R:

Author: ```suggestion
For example, rewriting the Python example from <https://github.com/pola-rs/polars#python> in R:
```

So that this goes straight to the example instead of the main page of the repo.

```{r}
library(neopolars) # TODO: rename to polars

df <- pl$DataFrame(
  A = 1:5,
  fruits = c("banana", "banana", "apple", "apple", "banana"),
  B = 5:1,
  cars = c("beetle", "audi", "beetle", "beetle", "beetle"),
)

> # embarrassingly parallel execution & very expressive query language

Author: ```suggestion
# very expressive query language
```

It is embarrassingly parallel, but it's just invisible to the user so it feels weird to mention it here.
